### ISSUE TITLE
Hyperparameter Optimization

### ISSUE THREAD

**Issue by truell20** [FidoProject/Fido#36]
It would be great if we could implement
- ~~Pruning of Neural Network architecture (PrunableLayer could be a subclass of Layer)~~
- Optimization of exploration level in the Fido control system
- ~~Learning constants could be optimized by implementing some of the trainers described in #26~~

**Comment by truell20** [comment]
Pruning and learning rate optimization done
